core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -969,435 +915,537 @@ private void syncLocalToRemote() {
 
-                final UpdateDescription localUpdateDescription =
-                        localChangeEvent.getUpdateDescription();
-                if (localUpdateDescription.getRemovedFields().isEmpty()
-                        && localUpdateDescription.getUpdatedFields().isEmpty()) {
-                  // if the translated update is empty, then this update is a noop, and we
-                  // shouldn't update because it would improperly update the version information.
-                  logger.debug(String.format(
-                          Locale.US,
-                          "t='%d': syncLocalToRemote ns=%s documentId=%s local change event "
-                                  + "update description is empty for UPDATE; dropping the event",
-                          logicalT,
-                          nsConfig.getNamespace(),
-                          docConfig.getDocumentId()));
-                  continue;
+                if (unprocessedEventVersionInfo != null) {
+                  final boolean unprocessedEventHasNoVersion =
+                      !unprocessedEventVersionInfo.hasVersion();
+                  final boolean localHasNoVersion = !localVersionInfo.hasVersion();
+
+                  @Nullable final DocumentVersionInfo.Version unprocessedEventVersion =
+                      unprocessedEventHasNoVersion ? null :
+                          unprocessedEventVersionInfo.getVersion();
+                  @Nullable final DocumentVersionInfo.Version localVersion =
+                      localHasNoVersion ? null : localVersionInfo.getVersion();
+
+                  if (!unprocessedEventHasNoVersion && !localHasNoVersion) {
+                    final boolean instanceIdMatch = localVersion.getInstanceId()
+                        .equals(unprocessedEventVersion.getInstanceId());
+                    final boolean lastSeenOlderThanRemote =
+                        unprocessedEventVersion.getVersionCounter()
+                            >= localVersion.getVersionCounter();
+
+                    final boolean hasCommittedVersion = instanceIdMatch
+                        && localVersion.getSyncProtocolVersion() == SYNC_PROTOCOL_VERSION
+                        && !lastSeenOlderThanRemote;
+
+                    if (!hasCommittedVersion) {
+                      action = SyncAction.CONFLICT;
+                      message = SyncMessage.VERSION_DIFFERENT_UNPROCESSED_EVENT_MESSAGE;
+                    }
+                  }
                 }
+                // 2. Otherwise, the unprocessed event can be safely dropped and ignored in future
+                // R2L passes. Continue on to checking the operation type.
+              }
 
+              if (action == null) { // if we haven't encountered an error/conflict already
+                // iii. Check the operation type
+                switch (localChangeEvent.getOperationType()) {
+                  // 1. INSERT
+                  case INSERT: {
+                    nextVersion =
```

Comment:
[minor] looks like you're still computing and sending the hash as part of the remote version

Reply:
fixed